examples/jsm/nodes/shadernode/ShaderNode.js

```diff
@@ -147,31 +148,43 @@ const ShaderNodeImmutable = function ( NodeClass, ...params ) {
 
 };
 
-const ShaderNodeScript = function ( jsFunc ) {
+class ShaderNodeInternal extends Node {
+
+	constructor( jsFunc ) {
 
-	// @TODO: Move this to Node extended class
+		super();
 
-	const self = {
+		this._jsFunc = jsFunc;
 
-		build: ( builder ) => {
+	}
 
-			self.call( {}, builder );
+	call( inputs, builder ) {
 
-			return '';
+		inputs = nodeObjects( inputs );
+
+		return nodeObject( this._jsFunc( inputs, builder ) );
+
+	}
 
-		},
+	build( builder ) {
```

Comment:
Maybe this method can be simplified by using `construct` or `generate`, but I'm not sure how...

Reply:
It's ok for now, the next step to this is allowing that `ShaderNode` can create function like this: https://github.com/mrdoob/three.js/pull/23350